test/e2e/framework/autoscaling/autoscaling_utils.go

```diff
@@ -495,6 +551,10 @@ func runServiceAndWorkloadForResourceConsumer(c clientset.Interface, ns, name st
 	}, metav1.CreateOptions{})
 	framework.ExpectNoError(err)
 
+	if sidecarStatus == Enable && sidecarType == Busy {
+		name = name + "-sidecar"
+	}
```

Comment:
From my understanding of the code, this would make the application idle and the sidecar busy right?

If that's correct, I wonder if making the sidecar busy should be decoupled since we basically setup the whole application to be run, but just before being run, it is replaced by the sidecar.
A suggestion would be to add a `runServiceAndSidecarForResourceConsumer` function that would only create the sidecar service and run the resource consumer for the sidecar. This function would be called only if `sidecarStatus == Enable && sidecarType == Busy`.
On a similar note, again to decouple applications and sidecars, we could create a `newSidecarResourceConsumer` function. wdyt?

Reply:
I think that is good idea. I was also thinking to decouple these logics with new functions.

Reply:
Initially I had created different functions but there was lot of duplications across both functions, so I added that in same function.

Reply:
I have refactored the code to move sidecar dependencies in different functions. Please have a look

Reply:
Thanks for looking into that, I really like how your refactored the code :+1: